UnitTest/UnitTest_CurrentRanges.c

```diff
@@ -0,0 +1,60 @@
+#include "../CurrentRanges.c"
+#include "mocks.h"
+#include <assert.h>
+
+static void TC_Testcase_001(void)
```

Comment:
The name of the test must be a statement of functionality.
Here, one assumption is that the input is sorted in ascending order.
Be alert to recognise assumptions/decisions explicitly.
Your list below is a bit shallow.

Reply:
my bad .
Worst ways of coding